### ISSUE TITLE
What is the purpose of "extern crate alloc"?

### ISSUE THREAD

**Issue by Golden-Pigeon** [Alignof/raki#22]
Hi, @Alignof

Thanks for your wonderful opensource work!

In [lib.rs](https://github.com/Alignof/raki/blob/0bc292ad7c899dc01d249871c007b5f0f19b47e3/src/lib.rs#L29), it declares a crate "alloc", which it is not used in the library. When I use raki in my bare-metal environment and compile the code, it reports errors
```
error: no global memory allocator found but one is required; link to std or add `#[global_allocator]` to a static item that implements the GlobalAlloc trait
```
but I don't want to implement an allocator or use a third party one.

Could you make a compilation condition to deside whether declare crate alloc or simply remove it?

**Comment by Alignof** [comment]
Dear @Golden-Pigeon 
Thanks for pointing out the problem.
The problem line was due to an error in my review of a past pull request and could simply be removed.
I will release a new version with the `extern crate alloc` removed and let you know if there are any problems.

**Comment by Alignof** [comment]
I have now released version 1.0.1 with the fixes.
https://crates.io/crates/raki

**Comment by Golden-Pigeon** [comment]
Thank you @Alignof 

I've tested my code with new version of raki now and there is nothing wrong now.

I will close the issue.
